products/schema.json

```
@@ -0,0 +1,530 @@
+      "title": "Version command",
+      "description": "Command to fetch the products version.",
+      "examples": [
+        "helm version",
+        "go version"
+      ],
+      "type": "string"
+    },
+    "releasePolicyLink": {
+      "title": "Release policy link",
+      "description": "URL to the producs release policy page.",
+      "type": "string",
+      "pattern": "^https?://.+$"
+    },
+    "iconUrl": {
+      "title": "Icon url",
+      "description": "URL to the product icon.",
+      "type": "string",
+      "format": "uri-template"
+    },
+    "category": {
+      "title": "Category",
+      "description": "Specify the category of this product.",
+      "enum": [
+        "app",
+        "db",
+        "device",
+        "framework",
+        "lang",
+        "library",
+        "os",
+        "server-app",
+        "service",
+        "standard"
+      ]
+    },
+    "permalink": {
+      "title": "Permalink",
+      "description": "The URL to this product. Must start with a slash.",
+      "examples": [
+        "/foo",
+        "/bar"
+      ],
+      "$ref": "#/$defs/link"
+    },
+    "tags": {
+      "title": "Tags",
+      "description": "Tags which fit this product. Only lowercase, numbers and minus are allowed. Separated via spaces.",
+      "type": "string",
+      "pattern": "^[a-z0-9\\-]+( [a-z0-9\\-]+)*$",
+      "examples": [
+        "foo bar"
+      ]
+    },
+    "alternate_urls": {
+      "title": "Alternate URLs",
+      "description": "Array of alternate urls for this product. Must start with a slash.",
+      "type": "array",
+      "items": {
+        "$ref": "#/$defs/link"
+      }
+    },
+    "auto": {
+      "type": "array",
+      "items": {
+        "oneOf": [
+          {
+            "type": "object",
+            "properties": {
+              "git": {
+                "title": "Git",
+                "description": "URL to git repo. Fetches releases from tags.",
+                "type": "string"
+              },
+              "regex": {
+                "$ref": "#/$defs/regex"
+              },
+              "template": {
+                "$ref": "#/$defs/template"
+              }
+            },
+            "required": [
+              "git"
+            ],
+            "additionalProperties": false
+          },
+          {
+            "type": "object",
+            "properties": {
+              "npm": {
+                "title": "Npm",
+                "description": "Name of the npm package to track.",
+                "type": "string"
+              },
+              "regex": {
+                "$ref": "#/$defs/regex"
+              },
+              "template": {
+                "$ref": "#/$defs/template"
+              }
+            },
+            "required": [
+              "npm"
+            ],
+            "additionalProperties": false
+          },
+          {
+            "type": "object",
+            "properties": {
+              "dockerhub": {
```

```suggestion
              "docker_hub": {
```

If recently migrated this auto method to python, and renamed it so we can have both for a short time. `dockerhub` is not used anymore, so it may be replaced by `docker_hub`.

```
amazon-linux.md: Schema validation errors were encountered.
  /dev/fd/63::$.auto[0]: {'docker_hub': 'library/amazonlinux', 'regex': '^(?:\\d+(\\.\\d+){2,4})$'} is not valid under any of the given schemas
  Underlying errors caused this.
```

---

ok, I adjusted this